### DISCUSSION TITLE
feat: add page count for tables

### DISCUSSION THREAD

**Discussion by Morphclue** [charmbracelet/bubbles#330] [General]
# Context
In my opinion, the current presentation style of the table does not clearly indicate the presence of other elements. As a user, I would assume that the table only contains seven items and comes to an end. My suggestion is to add a footer to the table, similar to lists, which displays a count of the total number of items, such as `<current position>/<total count>`. This would also allow for navigation through "pages" by using the arrow keys to increment the current position + the height of the table.

![charm-table-example](https://camo.githubusercontent.com/643fb5656286e55d13322c2361825ccef14786de4023d7610704a64348f8b00a/68747470733a2f2f73747566662e636861726d2e73682f627562626c65732d6578616d706c65732f7461626c652e676966)

## Alternatives
Another alternative that indicates that there are more than 7 elements in the list would be to already move the list up by one after moving down by one. This behaviour would stop when the bottom of the list would be reached.

**Comment by bashbunni** [comment]
Looking at the pagination example, this should already be possible. It looks like the [paginator](https://github.com/charmbracelet/bubbles/blob/master/paginator/paginator.go) is only taking in `ints` as arguments, so it doesn't depend on anything specific from the `list.Model`. [Here](https://github.com/charmbracelet/bubbletea/blob/master/examples/paginator/main.go) is an example of how to use the paginator bubble. I think you should be able to calculate the list of items you need for each page the same way they're doing it in the list. Bear in mind, I haven't tried this myself, but looking at the source code, I don't see why it wouldn't work.

**Reply by Morphclue** [reply]
Whoops. I seem to have overlooked the paginator in the examples. Wouldn't it still make sense to include the paginator in the table example? All three lists [list-simple](https://github.com/charmbracelet/bubbletea/blob/master/examples/list-simple/main.go), [list-default](https://github.com/charmbracelet/bubbletea/blob/master/examples/list-simple/main.go) and [list-fancy](https://github.com/charmbracelet/bubbletea/blob/master/examples/list-fancy/main.go) have this behavior built in by default.
